File: RepoTransBench/source_projects/Python/tomasbasham_ratelimit/tests/unit/exception_test.py
import pytest
from ratelimit.exception import RateLimitException

def test_ratelimit_exception_message_and_period():
    e = RateLimitException("limit reached", 4.5)
    assert isinstance(e, RateLimitException)
    assert e.period_remaining == 4.5
    assert str(e) == "limit reached"

File: RepoTransBench/target_projects/Python/Java/tomasbasham_ratelimit/src/test/java/com/tomasbasham/ratelimit/original/ExceptionTest.java
package com.tomasbasham.ratelimit.original;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

import com.tomasbasham.ratelimit.exception.RateLimitException;

public class ExceptionTest {
    @Test
    void testRateLimitExceptionMessageAndPeriod() {
        RateLimitException e = new RateLimitException("limit reached", 4.5);
        assertTrue(e instanceof RateLimitException);
        assertEquals(4.5, e.getPeriodRemaining(), 1e-9);
        assertEquals("limit reached", e.getMessage());
    }
}